packages/cashc/src/generation/GenerateTargetTraversal.ts

```diff
@@ -441,71 +330,31 @@ export default class GenerateTargetTraversal extends AstTraversal {
     return node;
   }
 
-  needsToVerifyCovenant(node: FunctionCallNode): boolean {
-    if (node.identifier.name !== GlobalFunction.CHECKSIG) return false;
-    if (!this.isCheckSigVerify) return false;
-    if (!this.covenantNeedsToBeVerified) return false;
-    if (this.scopeDepth > 0) return false;
-    return true;
-  }
-
-  verifyCovenant(): void {
-    // Duplicate [s, pk] that are on stack
-    this.emit(Op.OP_2DUP);
-    this.pushToStack('(value)');
-    this.pushToStack('(value)');
-
-    // Turn sig into datasig
-    this.emit([Op.OP_SWAP, Op.OP_SIZE, Op.OP_1SUB, Op.OP_SPLIT, Op.OP_DROP]);
-
-    // Retrieve preimage from stack and hash it
-    const preimageIndex = this.getStackIndex('$preimage');
-    this.removeFromStack(preimageIndex);
-    this.emit(encodeInt(preimageIndex));
-    this.emit(Op.OP_ROLL);
-    this.emit(Op.OP_SHA256);
-    this.pushToStack('(value)');
-
-    // Order arguments and perform OP_CHECKDATASIGVERIFY
-    this.emit(Op.OP_ROT);
-    this.emit(Op.OP_CHECKDATASIGVERIFY);
-    this.popFromStack(3);
-
-    this.covenantNeedsToBeVerified = false;
-  }
-
   visitInstantiation(node: InstantiationNode): Node {
-    if (node.identifier.name === Class.OUTPUT_P2PKH) {
-      // <output amount>
-      this.visit(node.parameters[0]);
-      // <VarInt 25 bytes> OP_DUP OP_HASH160 OP_PUSH<20>
-      this.emit(hexToBin('1976a914'));
-      this.emit(Op.OP_CAT);
+    if (node.identifier.name === Class.LOCKING_BYTECODE_P2PKH) {
+      // OP_DUP OP_HASH160 OP_PUSH<20>
+      this.emit(hexToBin('76a914'));
+      this.pushToStack('(value)');
       // <pkh>
-      this.visit(node.parameters[1]);
+      this.visit(node.parameters[0]);
```

Comment:
Just making sure I understand - this change is because value is no longer in the parameters and it's only the locking code, so it moves to position 0 in every case. Is that right?

Reply:
Exactly.